### DISCUSSION TITLE
Newbie question about flow networks

### DISCUSSION THREAD

**Discussion by petgriffin** [networkx/networkx#4902] [Q&A]
Hi
I am trying to model a gas pipeline network with multiple sources and a single sink.  Each node has a pressure and the flowrate between nodes is described by a look-up table (represented as an object), with the rate depending on the inlet and outlet node pressures, which vary as the flowrate through the network varies.  I need to calculate the maximum flow (with the sink node pressure fixed) and the sink node pressure (with the outlet flow fixed).  I had a look at the NetworkX documentation but am a little overwhelmed, so would be grateful if someone could point me in the right direction, especially with implementing the pressure drop.  Thanks in advance!

**Comment by dschult** [comment]
We have functions to find the maximum flow given demands and supplies.  Can you look at the documentation for the flow functions and see if the language there can map onto the language you are using and the situation you have?

**Comment by petgriffin** [comment]
Hi Dan

I had a look at the "Flows" algorithms in the user manual but I don't 
think any of them apply.  I have a maximum flow problem (rather than a 
shortest/quickest path problem). The NetworkX methods all seem to assume 
a fixed edge flow capacity, but in my case the edge flow is a non-linear 
function of the upstream and downstream node values; I use a look-up 
table to link the edge flows and node values.

Physically, I am looking at the flow of gas in a network.  Some other 
features of the problem are:
(a) the graph is directed so that gas flows from a series of sources via 
the network to a sink (currently one sink, but more than one in the 
general case)
(b) flow into the source nodes depends on the source node pressure
(c) flow out of the sink node is constrained either by a maximum rate or 
a minimum pressure (both defined)

I have managed to kludge a solution to the problem using  a SCIPY 
non-linear solver but am still testing it for robustness.

Best regards

Alun

On 29/07/2021 02:06, Dan Schult wrote:
>
> We have functions to find the maximum flow given demands and supplies. 
> Can you look at the documentation for the flow functions and see if 
> the language there can map onto the language you are using and the 
> situation you have?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/networkx/networkx/discussions/4902#discussioncomment-1084905>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALG2VBV4DS27YDTGM24XVODT2CSQBANCNFSM46XBSUBQ>.
>



-- 
This email has been checked for viruses by AVG.
https://www.avg.com

**Reply by dschult** [reply]
It looks like your problem has too much nonlinear feedback for our methods to work.
Of course, solving the full Navier-Stokes flow is the real problem you could solve, but hopefully the simplification of a network of pipes can be useful along with the Scipy solver.
